src/TestExplorer.ts

```diff
@@ -0,0 +1,153 @@
+            this.notebooksAccountKey = value;
+            break;
+          case TestExplorerParams.notebooksAccountSubscriptonId:
+            this.notebooksAccountSubscriptonId = value;
+            break;
+          case TestExplorerParams.notebooksAccountResourceGroup:
+            this.notebooksAccountResourceGroup = value;
+            break;
+        }
+      });
+  };
+
+  private handleMessage(event: MessageEvent) {
+    if (event.data.type === MessageTypes.InitTestExplorer || event.data.type === MessageTypes.HideConnectScreen) {
+      this.sendMessageToExplorerFrame(event.data);
+    }
+  }
+
+  private async AADLogin(): Promise<string> {
+    const credentials = new ClientSecretCredential(
+      this.notebooksTestRunnerApplicationId,
+      this.notebooksTestRunnerClientId,
+      this.notebooksTestRunnerClientSecret
+    );
+    const token = await credentials.getToken("https://management.core.windows.net/.default");
+    return token.token;
+  }
+
+  private async getDatabaseAccount(token: string): Promise<DatabaseAccountsGetResponse> {
+    const client = new CosmosDBManagementClient(new CustomSigner(token), this.notebooksAccountSubscriptonId);
+    return await client.databaseAccounts.get(this.notebooksAccountResourceGroup, this.notebooksAccountName);
+  }
+
+  private async initTestExplorer(): Promise<void> {
+    this.parseUrlParams();
+    const token = await this.AADLogin();
+    const databaseAccount = await this.getDatabaseAccount(token);
+
+    const content = {
+      type: MessageTypes.InitTestExplorer,
+      inputs: {
+        databaseAccount: databaseAccount,
+        subscriptionId: this.notebooksAccountSubscriptonId,
+        resourceGroup: this.notebooksAccountResourceGroup,
+        authorizationToken: `Bearer ${token}`,
+        features: {},
+        hasWriteAccess: true,
+        csmEndpoint: "https://management.azure.com",
+        dnsSuffix: "documents.azure.com",
+        serverId: "prod1",
+        extensionEndpoint: "/proxy",
+        subscriptionType: 3,
+        quotaId: "Internal_2014-09-01",
+        addCollectionDefaultFlight: "2",
+        isTryCosmosDBSubscription: false,
+        masterKey: this.notebooksAccountKey,
+        loadDatabaseAccountTimestamp: 1604663109836,
+        dataExplorerVersion: "1.0.1",
+        sharedThroughputMinimum: 400,
+        sharedThroughputMaximum: 1000000,
+        sharedThroughputDefault: 400,
+        defaultCollectionThroughput: {
+          storage: "100",
+          throughput: { fixed: 400, unlimited: 400, unlimitedmax: 100000, unlimitedmin: 400, shared: 400 }
+        },
+        flights: ["mongoindexeditor", "settingsv2"]
+      } as ViewModels.DataExplorerInputsFrame
+    };
+    window.postMessage(content, window.location.href);
+
+    const hideConnectContent = {
+      type: MessageTypes.HideConnectScreen
+    };
+    window.postMessage(hideConnectContent, window.location.href);
+  }
+
+  private sendMessageToExplorerFrame(data: unknown): void {
```

Comment:
Same as other file. Unclear why this is needed if loading in Portal mode.

Reply:
changed to "platform=Portal" and removed this.